### PR TITLE
Add @develop to ufs-weather-model

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR makes significant revisions to the ufs-weather-model package. Namely, it adds a version for the head of the develop branch and has updates to variants and dependencies, as well as one or two patches.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ufs-weather-model/package.py [hunk 1]
```diff
@@ -17,58 +19,129 @@ class UfsWeatherModel(CMakePackage):
...
+        "ccpp_32bit",
+        default=False,
+        description="Enable CCPP_32BIT (single precision arithmetic in slow physics)",
+    )
+    variant("avx2", default=True, description="Enable AVX2 instruction set")
+    variant(
+        "simdmultiarch", default=False, description="Enable multi-target SIMD instruction sets"
+    )
+    variant("debug", default=False, description="Enable DEBUG mode", when="@develop")
+    variant(
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/ufs-weather-model/package.py:42]: Note that this horrible debug link mpi option needs to go away, it's not portable and not something we should be doing. @aerorahul @DusanJovic-NOAA and I have fought this bad practice for a while, but with moving to spack-stack we should really get rid of it.

File: var/spack/repos/builtin/packages/ufs-weather-model/package.py [hunk 2]
```diff
@@ -17,58 +19,129 @@ class UfsWeatherModel(CMakePackage):
...
+        default=True,
+        description="Enable linkmpi option when DEBUG mode is on",
+        when="@develop",
+    )
+    variant("inline_post", default=False, description="Enable inline post")
+    variant("multi_gases", default=False, description="Enable multi gases in physics routines")
+    variant("moving_nest", default=False, description="Enable moving nest code", when="@develop")
+    variant("openmp", default=True, description="Enable OpenMP")
+    variant("parallel_netcdf", default=True, description="Enable parallel NetCDF")
+    variant(
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/ufs-weather-model/package.py:53]: I don't know if this option works at all, it's certainly not the way we integrate the UFS with JEDI here: https://github.com/JCSDA/ufs-bundle/pull/9

@mark-a-potts @aerorahul Can you provide more information on this option?

File: var/spack/repos/builtin/packages/ufs-weather-model/package.py [hunk 3]
```diff
@@ -17,58 +19,129 @@ class UfsWeatherModel(CMakePackage):
...
+        description="Enable JEDI as top level driver",
+        when="@develop",
+    )
+    variant(
+        "cmeps_aoflux",
+        default=False,
+        description="Enable atmosphere-ocean flux calculation in mediator",
+        when="@develop",
+    )
+    variant(
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/ufs-weather-model/package.py:65]: Why is this a variant? Is there a way to build the UFS without CCPP physics?

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji For what it's worth, my overall approach has been to try to make variants match as closely as possible with what's provided in cmake (though I have not implemented every available switch from the various submodules, only the one or two that I seemed to need for doing RTs), and leaving the rest up to the user. So nothing here is necessarily a guarantee of the underlying functionality in UFS WM.

**Review by climbfuji — APPROVED** [review]
I assume you have tested those changes, I haven't. It looks ok to me, though I don't think the `CCPP=false` option will work.

**Pull request merged**